packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -821,12 +822,12 @@ void main() {
       for (final Selectable selectable in (paragraph.registrar! as TestSelectionRegistrar).selectables) {
         selectable.dispatchSelectionEvent(
           SelectionEdgeUpdateEvent.forStart(
-            globalPosition: paragraph.getOffsetForCaret(start, Rect.zero),
+            globalPosition: paragraph.getOffsetForCaret(start, Rect.zero) + const Offset(0, 5),
           ),
         );
         selectable.dispatchSelectionEvent(
           SelectionEdgeUpdateEvent.forEnd(
-            globalPosition: paragraph.getOffsetForCaret(end, Rect.zero),
+            globalPosition: paragraph.getOffsetForCaret(end, Rect.zero) + const Offset(0, 5),
```

Comment:
Why did you need to add that small y offset?

Reply:
originally it was selecting right at the edge which can select across line if there is multi line text. It was ok because there was not multi line tests in this file prior